packages/order-utils/src/order_state_utils.ts

```diff
@@ -336,25 +329,19 @@ export class OrderStateUtils {
     ): Promise<ObjectMap<BigNumber>> {
         const decodedAssetData = assetDataUtils.decodeAssetDataOrThrow(assetData);
         let allowances: ObjectMap<BigNumber> = { ...initialAllowances };
-        switch (decodedAssetData.assetProxyId) {
-            case AssetProxyId.ERC20:
-            case AssetProxyId.ERC721:
-                const allowance = await this._balanceAndProxyAllowanceFetcher.getProxyAllowanceAsync(
-                    assetData,
-                    traderAddress,
-                );
-                const tokenAddress = (decodedAssetData as SingleAssetData).tokenAddress;
-                allowances[tokenAddress] = _.isUndefined(initialAllowances[tokenAddress])
-                    ? allowance
-                    : allowances[tokenAddress].add(allowance);
-                break;
-            case AssetProxyId.MultiAsset:
-                for (const assetDataElement of (decodedAssetData as MultiAssetData).nestedAssetData) {
-                    allowances = await this._getAssetBalancesAsync(assetDataElement, traderAddress, allowances);
-                }
-                break;
-            default:
-                throw new Error(`Proxy with id ${decodedAssetData.assetProxyId} not supported`);
+        if (assetDataUtils.isERC20AssetData(decodedAssetData) || assetDataUtils.isERC721AssetData(decodedAssetData)) {
+            const allowance = await this._balanceAndProxyAllowanceFetcher.getProxyAllowanceAsync(
+                assetData,
+                traderAddress,
+            );
+            const tokenAddress = (decodedAssetData as SingleAssetData).tokenAddress;
```

Comment:
You don't need this `as SingleAssetData` type assertion any more since we switched to using type guards.